README.md

```diff
@@ -1,65 +1,112 @@
+  implementation
+- [NetObserv](https://github.com/netobserv): Open Source Operator for network
+  observability 
+
+### Challenges for BPF in Kubernetes
+
+- Requires privileged pods
+  - BPF-enabled apps require at least CAP_BPF permissions, and more (depending
+    on what they need to do).
+  - These permissions give the pods great powers.
+- Handling multiple BPF programs
+  - Not all libraries support multiple XDP programs sharing the same interface.
+  - When multiple BPF programs are used, they can interfere with each other in
+    unpredictable ways.
+- Debugging BPF-related problems is difficult -- particularly on Kubernetes.
+- There are no standard libraries for loading and managing BPF programs leaving
+  each application developer to implement their own.
+- BPF byte code often needs to be compiled into associated applications.
+
+### What is bpfd?
+
+bpfd is a system daemon that supports, loading, unloading, modifying, and
+monitoring BPF programs. The solution for Kubernetes uses an operator, built
+with the Operator SDK, that can deploy the bpfd daemon and a bpfd agent to all
+the nodes in a cluster. The agent supports well-defined Kubernetes APIs (CRDs)
+that can be used to manage the lifecycle of BPF programs across a Kubernetes
+cluster.
+
+bpfd is developed in Rust and built on top of Aya, a Rust BPF library.
+
+The benefits of this solution include the following:
+
+- Security
+  - Improved security because only the bpfd daemon, which can be tightly
+    controlled, has the privileges needed to load BPF programs, while access to
+    the API can be controlled via standard RBAC methods.
+  - Gives the administrators control over who can load programs and allows
+    them to define rules for the ordering of networking BPF programs
+- Visibility/Debuggability
+  - Improved visibility into what BPF programs are running on a system, which
+    enhances the debuggability for developers, admins, and customer support.
+  - The greatest benefit is achievd when all apps use bpfd, but even if they
+    aren't bpfd can provide visibility into all the BPF programs loaded on the
+    nodes in a cluster.
+- Multi-program Support
+  - Support for the coexistence of multiple BPF programs from multiple users.
+  - Uses the [libxdp multiprog
+    protocol](https://github.com/xdp-project/xdp-tools/blob/master/lib/libxdp/protocol.org)
+    to allow multiple XDP programs on single interface
+  - This same protocol is also supported for TC programs to provide a common
+    multi-program user experience across both TC and XDP.
+  - An additional benefit for TC is that it makes the re-ordering and inserting
+    of new programs atomic.
+- Productivity
+  - Simplifies the deployment of BPF programs in a Kubernetes cluster.
+  - Improved ease of use and productivity because developers can use a standard
+    framework for managing BPF programs rather than developing their own.
+  - Developers can still use Cilium/libxdp/Aya/etc libraries for BPF
```

Comment:
s/libxdp/libbpf/g

Reply:
I changed this, but
a) can't they use libxdp too?
b) I didn't do /g because the other instances were where I mentioned libxdp multiprog.

Reply:
You could, but I don't feel its a good example:

libxdp is used for:
a. loading and attaching xdp programs
b. working with af_xdp sockets

if you move (a) to bpfd, then you aren't left with much

Reply:
I'll leave libxdp out to keep things simple, but I feel like it's disingenuous for us to say you can't attach multiple programs to an XDP hookpoint without bpfd because you can -- you just need to use libxdp.

Reply:
Never mind that comment I just made and deleted.  I agree with you.